src/matrixHelpers/FormioLoader/index.js

```diff
@@ -0,0 +1,202 @@
+  }
+  const bodyContainer = $("body");
+  const defaultRedirect = "contact-us/response/";
+  /*
+   * setup config
+   */
+  const baseUrl = `https://${opts.envUrl.trim()}`;
+  const formioContainerId = formioElem.getAttribute("id");
+  const submitBtn = $(`#${formioContainerId} button[name='data[submit]']`);
+  let formName = "";
+  // Check if value is true/exists and is numeric
+  if (opts.form_revision) {
+    formName = `${opts.formName}/v/${opts.form_revision}`;
+  } else {
+    formName = opts.formName;
+  }
+  const { projectName } = opts;
+  const formConfirmation = opts.formConfirmation || defaultRedirect;
+  const namespace = opts.namespace || `formio-${projectName}`;
+  const formUrl = `${baseUrl}/${projectName}/${formName}`;
+
+  /*
+   * init formio instance
+   */
+  const formio = new Formio(formUrl, {
+    base: baseUrl,
+    project: `${baseUrl}/${projectName}`,
+    namespace,
+  });
+  formioElem.dataset.formio = JSON.stringify(formio);
+  formioElem.dataset.formioFormUrl = formUrl;
+
+  /*
+   * load formio form
+   */
+  Formio.createForm(
+    formioElem,
+    formUrl,
+    // form,
+    {
+      // Turn off default buttons
+      buttonSettings: {
+        showCancel: false,
+        showPrevious: false,
+        showNext: false,
+        showSubmit: false,
+      },
+      i18n: {
+        en: { pattern: "Must use the format shown" },
+      },
+      formio,
+      namespace: formio.options.namespace,
+    }
+  ).then((wizard) => {
+    wizard.formio = formio;
+    wizard.options.formio = formio;
+
+    // eslint-disable-next-line no-underscore-dangle
+    const formTitle = wizard._form.title;
+    // eslint-disable-next-line no-underscore-dangle
+    const formModified = wizard._form.modified;
+
+    // Force new tab on formlinks
+    bodyContainer.on("click", `#${formioContainerId} a`, (e) => {
+      e.target.target = "_blank";
+    });
+
+    // Change event/GTM
+    wizard.on("click", (change) => {
+      const changeObj = change;
+      if (
+        typeof changeObj.changed !== "undefined" &&
+        typeof changeObj.changed.component !== "undefined"
+      ) {
+        window.dataLayer.push({
+          event: "formio-interaction",
+          "formio-name": formTitle,
+          "formio-input-id": changeObj.changed.component.id,
+          "formio-input-type": changeObj.changed.component.type,
+          "formio-input-value": changeObj.changed.component.value,
+          "formio-input-key": changeObj.changed.component.key,
+          "formio-input-label-raw": changeObj.changed.component.label,
+          "formio-version": formModified,
+          "formio-category": `Form: ${formTitle}`,
+          "formio-action": "Value changed",
+        });
+      }
+    });
+
+    // Must use 'applicationSubmit' custom event on primary submit
+    wizard.on("applicationSubmit", () => {
+      submitBtn.attr("disabled", true);
+      wizard
+        .submit()
+        .then(() => {
+          if (formConfirmation) {
```

Comment:
hmm, if confirmation url is given then use it, if not set, then don't change the page. form.io form can have logic to give a thank you on said page.

this also breaks the model where this could be a 'model' over the top of a page in getting 'feedback star rating' and all it needed to do it give a thanks and close out. to alter the href of the page breaks that workflow.

Reply:
as well as being in the footer feedack which gives a 'thanks for the info' replacement of that div, not the entire page being refreshed.

Reply:
related to my previous comment, this custom event only fired if applicationSubmit event is defined in a form, so with normal form this part will not be executed

Reply:
Unsure about that, as I believe it is always included and is fired when submission is successful.

From the url part. I'd say this is related to the /contact-us on qld.gov.au

Reply:
mmm... i'll do some testing there, i didn't really like this piece of custom logic place in here, i guess it's better place in the custom logic in the form stored in the platform